### PR TITLE
Forwardport #786

### PR DESCRIPTION
**Pull request by MaximilianMeister**
https://github.com/crowbar/crowbar-core/pull/786

### CODE REVIEW

File: crowbar_framework/app/models/api/upgrade.rb [hunk 1]
```diff
@@ -29,35 +29,50 @@ def checks
...
+          ceph_status = Api::Crowbar.ceph_status
           ret[:ceph_healthy] = {
             required: true,
-            passed: ceph_healthy?,
-            errors: ceph_health_check_errors
+            passed: ceph_status.empty?,
+            errors: ceph_status.empty? ? {} : ceph_health_check_errors(ceph_status)
           } if Api::Crowbar.addons.include?("ceph")
+
+          ha_presence = Api::Crowbar.ha_presence_check
```
Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: I believe it's part of Api:Pacemaker in master branch

File: crowbar_framework/app/models/api/upgrade.rb [hunk 2]
```diff
@@ -29,35 +29,50 @@ def checks
...
           ret[:ha_configured] = {
             required: false,
-            passed: ha_present?,
-            errors: ha_presence_errors
+            passed: ha_presence.empty?,
+            errors: ha_presence.empty? ? {} : ha_presence_errors(ha_presence)
           }
+
+          clusters_health = Api::Crowbar.clusters_health_report
```
Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: Api:Pacemaker

File: crowbar_framework/app/models/api/upgrade.rb (hunk 1 above)
Comment (MaximilianMeister) [inline crowbar_framework/app/models/api/upgrade.rb]: thanks!

**Comment by MaximilianMeister** [comment]
@jsuchome @rsalevsky ping, I just rebased

File: crowbar_framework/app/models/api/upgrade.rb [hunk 3]
```diff
@@ -29,35 +29,50 @@ def checks
...
+          ha_presence = Api::Pacemaker.ha_presence_check
           ret[:ha_configured] = {
             required: false,
-            passed: ha_present?,
-            errors: ha_presence_errors
+            passed: ha_presence.empty?,
+            errors: ha_presence.empty? ? {} : ha_presence_errors(ha_presence)
           }
+
+          clusters_health = Api::Pacemaker.clusters_health_report
```
Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: I'ts `Api::Pacemaker.health_report`

Comment (MaximilianMeister) [inline crowbar_framework/app/models/api/upgrade.rb]: thanks. fixed

**Pull request merged**